sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -424,12 +433,3792 @@ public void createOptionsWithTimeToExpireRelativeToNow() {
         DataLakePathCreateOptions options = new DataLakePathCreateOptions()
             .setScheduleDeletionOptions(deletionOptions);
 
-        StepVerifier.create(fc.createWithResponse(options, null))
-            .assertNext(r -> assertEquals(201, r.getStatusCode()))
+        assertAsyncResponseStatusCode(fc.createWithResponse(options, null), 201);
+
+        StepVerifier.create(fc.getProperties())
+            .assertNext(r -> compareDatesWithPrecision(r.getExpiresOn(), r.getCreationTime().plusDays(6)))
+            .verifyComplete();
+    }
+
+    @Test
+    public void createIfNotExistsMin() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+        fc.createIfNotExists().block();
+
+        StepVerifier.create(fc.exists())
+            .expectNext(true)
+            .verifyComplete();
+    }
+
+    @Test
+    public void createIfNotExistsDefaults() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        StepVerifier.create(fc.createIfNotExistsWithResponse(new DataLakePathCreateOptions(), null))
+            .assertNext( r -> {
+                assertEquals(201, r.getStatusCode());
+                validateBasicHeaders(r.getHeaders());
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void createIfNotExistsOverwrite() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        assertAsyncResponseStatusCode(fc.createIfNotExistsWithResponse(new DataLakePathCreateOptions(), null),
+            201);
+
+        StepVerifier.create(fc.exists())
+            .expectNext(true)
+            .verifyComplete();
+
+        assertAsyncResponseStatusCode(fc.createIfNotExistsWithResponse(new DataLakePathCreateOptions(), null),
+            409);
+    }
+
+    @Test
+    public void createIfNotExistsExists() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+        fc.createIfNotExists().block();
+
+        assertTrue(fc.exists().block());
+    }
+
+    @ParameterizedTest
+    @CsvSource(value={"null, null, null, null, null","control, disposition, encoding, language, type"})
```

Comment:
Not sure if CSV source strips leading and trailing spaces

```suggestion
    @CsvSource(value={"null,null,null,null,null", "control,disposition,encoding,language,type"})
```